### PR TITLE
Add platform search

### PR DESCRIPTION
**Pull request by finestructure**
This adds support for `platform:ios,macos` style filters.

Merge after #1432

### CODE REVIEW

**Review by daveverwer — APPROVED** [review]
My only comment would be that the FAQ could use a bit more documentation on exactly how that filter works. See the sections below the bulleted list? I think we should have one for platform, focusing on how to do a single, or multiple platform searches and whether multiple platform search is an AND/OR operator.

I know when @Sherlouk proposed it, he proposed it to work as `platform:ios` `platform:macos` which would filter with an AND so if this syntax is an OR, that works, I guess? But it needs documenting.

**Comment by Sherlouk** [comment]
It reads to me as if both `platform:ios platform:macos` and `platform:ios,macos` would do the same thing with this PR? (Not against the comma separated syntax but would be good to add a small note to the FAQ)

**Comment by daveverwer** [comment]
We should also replace the `last_commit` filter suggestion with one for `platform` as that would make a great set of suggestions. Last maintenance activity, platform, and stars.

**Comment by finestructure** [comment]
> We should also replace the `last_commit` filter suggestion with one for `platform` as that would make a great set of suggestions. Last maintenance activity, platform, and stars.

How about we make that a follow-up?

File: Sources/App/Core/SearchFilter/Filters/PlatformSearchFilter.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            SQLRaw("@>"),
+            SQLBind(value)
+        )
+    }
+
+    func createViewModel() -> SearchFilterViewModel {
+        .init(
+            key: "platform compatibility",
+            comparison: comparison,
+            value: value.map(\.rawValue).joined(separator: ",")
```
[the remark this answers is not in the document]
Comment (daveverwer) [reply Sources/App/Core/SearchFilter/Filters/PlatformSearchFilter.swift:62]: My vote would be for option 4 here. We have a function to turn a list into a sentence like this.

**Comment by daveverwer** [comment]
> How about we make that a follow-up?

It's a trivial change. I can push it now.

File: Sources/App/Core/SearchFilter/Filters/PlatformSearchFilter.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (finestructure) [reply Sources/App/Core/SearchFilter/Filters/PlatformSearchFilter.swift:62]: I'm not a super fan of expanding this into forms that wouldn't work as input, so I'd prefer to use 1 or 2. Thoughts, @daveverwer ?

**Comment by finestructure** [comment]
Ah great, I thought this required JS fiddling.

File: Sources/App/Core/SearchFilter/Filters/PlatformSearchFilter.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (daveverwer) [reply Sources/App/Core/SearchFilter/Filters/PlatformSearchFilter.swift:62]: I feel quite strongly that the primary purpose of this area is to better explain the filter that people enter, so I'm afraid I feel the opposite. The original idea behind this was to make it into a more readable version of what the filter is doing.

[the remark this answers is not in the document]
Comment (daveverwer) [reply Sources/App/Core/SearchFilter/Filters/PlatformSearchFilter.swift:62]: The intention was never that it'd be copyable back to the filter.

**Comment by daveverwer** [comment]
> Ah great, I thought this required JS fiddling.

Technically, yes. But it's a bit of static JSON. See 91d6e9b28cb4695ccdec6a88abad358306119c1c

File: Resources/Scripts/search_filter_suggestions.js [hunk 2]
```diff
@@ -26,9 +26,9 @@ export class SPISearchFilterSuggestions {
         'Filter to packages having a commit or a closed/merged pull request or issue in the last three months.',
     },
     {
-      text: 'last commit',
-      filter: `last_commit:>${this.formattedFilterDate()}`,
-      description: 'Filter to packages having a commit in the last three months.',
+      text: 'compatible platforms',
+      filter: `platforms:ios,macos`,
```
Comment (Sherlouk) [inline Resources/Scripts/search_filter_suggestions.js:30]:
```suggestion
      filter: `platform:ios,macos`,
```

👀

Comment (daveverwer) [reply Resources/Scripts/search_filter_suggestions.js:30]: 😬

**Pull request merged**